fxa/oauth.py

```diff
@@ -239,11 +240,21 @@ def verify_token(self, token, scope=None):
             resp = None
 
         if resp is None:
-            url = '/verify'
-            body = {
-                'token': token
+            ## TODO we want to fetch https://oauth.accounts.firefox.com/.well-known/openid-configuration and then get the jwks_uri key to get the following url, but we'll just hardcode it like this for now until we are closer to production. We need to do this because oauth.accounts.firefox.com does not yet respond to .well-known, but accounts.firefox.com does; for consistency, oauth.accounts.firefox.com should respond to .well-known. 
+            
+            secret = jwt.algorithms.RSAAlgorithm.from_jwk(
+                self.apiclient.get('/v1/jwks')
+            )
+            decoded = jwt.decode(
+                token, secret, algorithms=['RS256'], options={'verify_aud': False}
+            )
+            resp = {
+                'user': decoded['sub'],
+                'client_id': decoded['client_id'],
+                'scope': decoded['scope'],
+                'generation': decoded['fxa-generation'],
+                'profile_changed_at': decoded['fxa-profileChangedAt']
             }
-            resp = self.apiclient.post(url, body)
```

Comment:
The FxA server may choose to issue JWT tokens or not at its discretion, so I think we need to keep the remote call to `/verify` as a fallback, particularly in a general-purpose library. What do you think about something like this?

```
try:
    decoded = jwt.algoritms.decode(token, secret, ...)
    resp = { ... }
except jwt.exceptions.DecodeError:
   resp = self.apiclient.post(`/verify`, {'token': token })
```

The `DecodeError` exception seems to be specific to the "I couldn't parse a proper JWT out of what you have me" error case.

Reply:
👍